### ISSUE TITLE
ICE in rustc: `index out of bounds: the len is 74 but the index is 74`

### ISSUE THREAD

**Issue by db48x** [rust-lang/rust#100306]
### Code

I haven’t actually tried to minimize this yet, but here is the new code that I wrote. It has some silly errors which I haven’t even tried to fix yet.

```Rust
#[instrument(skip(api))]
pub async fn get_list_members<Auth>(api: &TwitterApi<Auth>, rl: &DefaultRateLimiter, list_id: u64) -> Result<Vec<User>>
where Auth: Authorization + Send + Sync + std::fmt::Debug {
  let response = retry(||
    rl.until_ready().await;
    api.get_list_members(list_id)
      .user_fields([UserField::Id, UserField::Name])
      .max_results(100)
      .send()
      .await
      .map(|response| Some(response))
  );
  get_all_pages(rl, response, None)
    .await
    .with_context(|| format!("Failed to fetch list members for list {list_id}"))
}

type Response<T> = twitter_v2::Result<Option<ApiResponse<Auth, Vec<T>, ResultCountMeta>>>;

#[instrument(skip(func))]
pub async fn retry(rl: &DefaultRateLimiter, func: dyn Fn() -> Response<T>) {
  loop {
    rl.until_ready().await;
    let response = func();
    match response {
      Err(twitter_v2::Error::Request(ref e)) if e.is_timeout() || e.is_connect() || e.status() == Some(reqwest::StatusCode::TOO_MANY_REQUESTS) => {
        warn!(error=?e, "retrying due to request error");
      },
      Err(twitter_v2::Error::Api(ref e)) if e.status == reqwest::StatusCode::TOO_MANY_REQUESTS => {
        warn!(error=?e, "retrying due to 429 response");
      },
      Err(e) => return Err(anyhow!(e)),
      _ => {
        return response;
      }
    }
  }
}
```

### Meta

`rustc --version --verbose`:
```
rustc 1.64.0-nightly (6dbae3ad1 2022-07-25)
binary: rustc
commit-hash: 6dbae3ad19309bb541d9e76638e6aa4b5449f29a
commit-date: 2022-07-25
host: x86_64-unknown-linux-gnu
release: 1.64.0-nightly
LLVM version: 14.0.6
```

### Error output

```
error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.64.0-nightly (6dbae3ad1 2022-07-25) running on x86_64-unknown-linux-gnu

note: compiler flags: --crate-type lib -C embed-bitcode=no -C debuginfo=2 -C incremental -C target-cpu=znver2

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
#0 [typeck] type-checking `twitter::get_list_members`
#1 [mir_built] building MIR for `twitter::get_list_members`
#2 [unsafety_check_result] unsafety-checking `twitter::get_list_members`
#3 [mir_const] processing MIR for `twitter::get_list_members`
#4 [mir_promoted] processing `twitter::get_list_members`
#5 [mir_borrowck] borrow-checking `twitter::get_list_members`
#6 [type_of] computing type of `twitter::get_list_members::{opaque#0}`
#7 [check_mod_item_types] checking item types in module `twitter`
#8 [analysis] running analysis passes on this crate
end of query stack
Some errors have detailed explanations: E0061, E0308, E0412.
For more information about an error, try `rustc --explain E0061`.
warning: `anaphora` (lib) generated 1 warning
error: could not compile `anaphora` due to 7 previous errors; 1 warning emitted
```

<details><summary><strong>Backtrace</strong></summary>
<p>

```
thread 'rustc' panicked at 'index out of bounds: the len is 74 but the index is 74', /cargo/registry/src/github.com-1ecc6299db9ec823/ena-0.14.0/src/snapshot_vec.rs:199:10
stack backtrace:
   0:     0x7fce702857d0 - std::backtrace_rs::backtrace::libunwind::trace::h74e6731adcfb8adc
                               at /rustc/6dbae3ad19309bb541d9e76638e6aa4b5449f29a/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   1:     0x7fce702857d0 - std::backtrace_rs::backtrace::trace_unsynchronized::h22bd7e644b454809
                               at /rustc/6dbae3ad19309bb541d9e76638e6aa4b5449f29a/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7fce702857d0 - std::sys_common::backtrace::_print_fmt::hec626af75cdf5d14
                               at /rustc/6dbae3ad19309bb541d9e76638e6aa4b5449f29a/library/std/src/sys_common/backtrace.rs:66:5
   3:     0x7fce702857d0 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hfe4102e485da4cf8
                               at /rustc/6dbae3ad19309bb541d9e76638e6aa4b5449f29a/library/std/src/sys_common/backtrace.rs:45:22
   4:     0x7fce702dfa5c - core::fmt::write::h78ef6088ea150cd5
                               at /rustc/6dbae3ad19309bb541d9e76638e6aa4b5449f29a/library/core/src/fmt/mod.rs:1198:17
   5:     0x7fce70276eb5 - std::io::Write::write_fmt::hc440e7a0893fa000
                               at /rustc/6dbae3ad19309bb541d9e76638e6aa4b5449f29a/library/std/src/io/mod.rs:1672:15
   6:     0x7fce702884a1 - std::sys_common::backtrace::_print::h6f08412e91897b34
                               at /rustc/6dbae3ad19309bb541d9e76638e6aa4b5449f29a/library/std/src/sys_common/backtrace.rs:48:5
   7:     0x7fce702884a1 - std::sys_common::backtrace::print::he5a852bdc192e345
                               at /rustc/6dbae3ad19309bb541d9e76638e6aa4b5449f29a/library/std/src/sys_common/backtrace.rs:35:9
   8:     0x7fce702884a1 - std::panicking::default_hook::{{closure}}::hf308fab78552e6c7
                               at /rustc/6dbae3ad19309bb541d9e76638e6aa4b5449f29a/library/std/src/panicking.rs:295:22
   9:     0x7fce70288173 - std::panicking::default_hook::h66b19fed514ed5ed
                               at /rustc/6dbae3ad19309bb541d9e76638e6aa4b5449f29a/library/std/src/panicking.rs:314:9
  10:     0x7fce70aa6f44 - rustc_driver[45495022f976cfcf]::DEFAULT_HOOK::{closure#0}::{closure#0}
  11:     0x7fce70288c76 - std::panicking::rust_panic_with_hook::h9f418f85f12ababe
                               at /rustc/6dbae3ad19309bb541d9e76638e6aa4b5449f29a/library/std/src/panicking.rs:702:17
  12:     0x7fce70288ac7 - std::panicking::begin_panic_handler::{{closure}}::h0ed507e68a5b06bf
                               at /rustc/6dbae3ad19309bb541d9e76638e6aa4b5449f29a/library/std/src/panicking.rs:588:13
  13:     0x7fce70285cf4 - std::sys_common::backtrace::__rust_end_short_backtrace::h5dcdeed6df145b40
                               at /rustc/6dbae3ad19309bb541d9e76638e6aa4b5449f29a/library/std/src/sys_common/backtrace.rs:138:18
  14:     0x7fce702887f2 - rust_begin_unwind
                               at /rustc/6dbae3ad19309bb541d9e76638e6aa4b5449f29a/library/std/src/panicking.rs:584:5
  15:     0x7fce7024bb93 - core::panicking::panic_fmt::h722432ad1c1d9291
                               at /rustc/6dbae3ad19309bb541d9e76638e6aa4b5449f29a/library/core/src/panicking.rs:142:14
  16:     0x7fce7024bad2 - core::panicking::panic_bounds_check::h1a72a5767b9ed10b
                               at /rustc/6dbae3ad19309bb541d9e76638e6aa4b5449f29a/library/core/src/panicking.rs:84:5
  17:     0x7fce72a18be8 - <rustc_infer[fdb6242658469ba6]::infer::ShallowResolver as rustc_middle[64de8ef72bf26916]::ty::fold::TypeFolder>::fold_ty
  18:     0x7fce72a3e7d2 - <rustc_infer[fdb6242658469ba6]::infer::resolve::OpportunisticVarResolver as rustc_middle[64de8ef72bf26916]::ty::fold::FallibleTypeFolder>::try_fold_ty
  19:     0x7fce72a61aec - rustc_middle[64de8ef72bf26916]::ty::util::fold_list::<rustc_infer[fdb6242658469ba6]::infer::resolve::OpportunisticVarResolver, rustc_middle[64de8ef72bf26916]::ty::subst::GenericArg, <&rustc_middle[64de8ef72bf26916]::ty::list::List<rustc_middle[64de8ef72bf26916]::ty::subst::GenericArg> as rustc_middle[64de8ef72bf26916]::ty::fold::TypeFoldable>::try_fold_with<rustc_infer[fdb6242658469ba6]::infer::resolve::OpportunisticVarResolver>::{closure#0}>
  20:     0x7fce72a3e7fd - <rustc_infer[fdb6242658469ba6]::infer::resolve::OpportunisticVarResolver as rustc_middle[64de8ef72bf26916]::ty::fold::FallibleTypeFolder>::try_fold_ty
  21:     0x7fce72a23a7b - <rustc_middle[64de8ef72bf26916]::ty::Ty as rustc_middle[64de8ef72bf26916]::ty::fold::TypeSuperFoldable>::super_fold_with::<rustc_infer[fdb6242658469ba6]::infer::resolve::OpportunisticVarResolver>
  22:     0x7fce72a81f16 - <&rustc_middle[64de8ef72bf26916]::ty::list::List<rustc_middle[64de8ef72bf26916]::ty::subst::GenericArg> as rustc_middle[64de8ef72bf26916]::ty::fold::TypeFoldable>::try_fold_with::<rustc_infer[fdb6242658469ba6]::infer::resolve::OpportunisticVarResolver>
  23:     0x7fce72a3e7fd - <rustc_infer[fdb6242658469ba6]::infer::resolve::OpportunisticVarResolver as rustc_middle[64de8ef72bf26916]::ty::fold::FallibleTypeFolder>::try_fold_ty
  24:     0x7fce71a185d6 - <(rustc_middle[64de8ef72bf26916]::ty::Ty, rustc_middle[64de8ef72bf26916]::ty::Ty) as rustc_middle[64de8ef72bf26916]::ty::fold::TypeFoldable>::fold_with::<rustc_infer[fdb6242658469ba6]::infer::resolve::OpportunisticVarResolver>
  25:     0x7fce71a09b5f - <rustc_infer[fdb6242658469ba6]::infer::InferCtxt>::same_type_modulo_infer
  26:     0x7fce71a0f0a2 - <rustc_infer[fdb6242658469ba6]::infer::InferCtxt>::note_type_err
  27:     0x7fce71a059c7 - <rustc_infer[fdb6242658469ba6]::infer::InferCtxt>::report_and_explain_type_error
  28:     0x7fce70da1b8a - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::report_arg_errors
  29:     0x7fce72053805 - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_argument_types
  30:     0x7fce72029266 - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_call
  31:     0x7fce72036702 - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  32:     0x7fce720503a5 - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_argument_types
  33:     0x7fce72029266 - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_call
  34:     0x7fce72036702 - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  35:     0x7fce720664ac - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_match
  36:     0x7fce72037dfb - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  37:     0x7fce7203687f - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  38:     0x7fce720572bd - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_block_with_expected
  39:     0x7fce7203682f - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  40:     0x7fce720956ba - rustc_typeck[9089f5c415cb9e35]::check::check::check_fn
  41:     0x7fce72068c0e - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_expr_closure
  42:     0x7fce7203935a - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  43:     0x7fce720503a5 - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_argument_types
  44:     0x7fce72029266 - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_call
  45:     0x7fce72036702 - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  46:     0x7fce72056188 - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_decl
  47:     0x7fce72057190 - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_block_with_expected
  48:     0x7fce7203682f - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  49:     0x7fce72038359 - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  50:     0x7fce720572bd - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_block_with_expected
  51:     0x7fce7203682f - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  52:     0x7fce720956ba - rustc_typeck[9089f5c415cb9e35]::check::check::check_fn
  53:     0x7fce72068c0e - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_expr_closure
  54:     0x7fce7203935a - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  55:     0x7fce720503a5 - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_argument_types
  56:     0x7fce72029266 - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_call
  57:     0x7fce72036702 - <rustc_typeck[9089f5c415cb9e35]::check::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  58:     0x7fce720956ba - rustc_typeck[9089f5c415cb9e35]::check::check::check_fn
  59:     0x7fce7211f926 - <rustc_infer[fdb6242658469ba6]::infer::InferCtxtBuilder>::enter::<&rustc_middle[64de8ef72bf26916]::ty::context::TypeckResults, <rustc_typeck[9089f5c415cb9e35]::check::inherited::InheritedBuilder>::enter<rustc_typeck[9089f5c415cb9e35]::check::typeck_with_fallback<rustc_typeck[9089f5c415cb9e35]::check::typeck::{closure#0}>::{closure#1}, &rustc_middle[64de8ef72bf26916]::ty::context::TypeckResults>::{closure#0}>
  60:     0x7fce720a957f - rustc_typeck[9089f5c415cb9e35]::check::typeck
  61:     0x7fce7311c54f - <rustc_query_system[4a077ceeb940558c]::dep_graph::graph::DepGraph<rustc_middle[64de8ef72bf26916]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[64de8ef72bf26916]::ty::context::TyCtxt, rustc_span[ae34330c9831655e]::def_id::LocalDefId, &rustc_middle[64de8ef72bf26916]::ty::context::TypeckResults>
  62:     0x7fce72660bca - rustc_query_system[4a077ceeb940558c]::query::plumbing::try_execute_query::<rustc_query_impl[c278194eeae80ced]::plumbing::QueryCtxt, rustc_query_system[4a077ceeb940558c]::query::caches::DefaultCache<rustc_span[ae34330c9831655e]::def_id::LocalDefId, &rustc_middle[64de8ef72bf26916]::ty::context::TypeckResults>>
  63:     0x7fce72649238 - <rustc_query_impl[c278194eeae80ced]::Queries as rustc_middle[64de8ef72bf26916]::ty::query::QueryEngine>::typeck
  64:     0x7fce72b38945 - <rustc_middle[64de8ef72bf26916]::ty::context::TyCtxt>::typeck_opt_const_arg
  65:     0x7fce7229e445 - rustc_mir_build[2167d4e906bb5467]::build::mir_built
  66:     0x7fce731102ac - <rustc_query_system[4a077ceeb940558c]::dep_graph::graph::DepGraph<rustc_middle[64de8ef72bf26916]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[64de8ef72bf26916]::ty::context::TyCtxt, rustc_middle[64de8ef72bf26916]::ty::WithOptConstParam<rustc_span[ae34330c9831655e]::def_id::LocalDefId>, &rustc_data_structures[c1b71785fb17ed5f]::steal::Steal<rustc_middle[64de8ef72bf26916]::mir::Body>>
  67:     0x7fce726559fe - rustc_query_system[4a077ceeb940558c]::query::plumbing::try_execute_query::<rustc_query_impl[c278194eeae80ced]::plumbing::QueryCtxt, rustc_query_system[4a077ceeb940558c]::query::caches::DefaultCache<rustc_middle[64de8ef72bf26916]::ty::WithOptConstParam<rustc_span[ae34330c9831655e]::def_id::LocalDefId>, &rustc_data_structures[c1b71785fb17ed5f]::steal::Steal<rustc_middle[64de8ef72bf26916]::mir::Body>>>
  68:     0x7fce726479d5 - <rustc_query_impl[c278194eeae80ced]::Queries as rustc_middle[64de8ef72bf26916]::ty::query::QueryEngine>::mir_built
  69:     0x7fce71f6fe97 - rustc_mir_transform[b0e151ac016bcafc]::check_unsafety::unsafety_check_result
  70:     0x7fce71f6c2fe - <rustc_mir_transform[b0e151ac016bcafc]::check_unsafety::provide::{closure#0} as core[807c2f8b07026ed8]::ops::function::FnOnce<(rustc_middle[64de8ef72bf26916]::ty::context::TyCtxt, rustc_span[ae34330c9831655e]::def_id::LocalDefId)>>::call_once
  71:     0x7fce7311d05f - <rustc_query_system[4a077ceeb940558c]::dep_graph::graph::DepGraph<rustc_middle[64de8ef72bf26916]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[64de8ef72bf26916]::ty::context::TyCtxt, rustc_span[ae34330c9831655e]::def_id::LocalDefId, &rustc_middle[64de8ef72bf26916]::mir::query::UnsafetyCheckResult>
  72:     0x7fce72662c4a - rustc_query_system[4a077ceeb940558c]::query::plumbing::try_execute_query::<rustc_query_impl[c278194eeae80ced]::plumbing::QueryCtxt, rustc_query_system[4a077ceeb940558c]::query::caches::DefaultCache<rustc_span[ae34330c9831655e]::def_id::LocalDefId, &rustc_middle[64de8ef72bf26916]::mir::query::UnsafetyCheckResult>>
  73:     0x7fce72648e3e - <rustc_query_impl[c278194eeae80ced]::Queries as rustc_middle[64de8ef72bf26916]::ty::query::QueryEngine>::unsafety_check_result
  74:     0x7fce71f4bbd0 - rustc_mir_transform[b0e151ac016bcafc]::mir_const
  75:     0x7fce731102ac - <rustc_query_system[4a077ceeb940558c]::dep_graph::graph::DepGraph<rustc_middle[64de8ef72bf26916]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[64de8ef72bf26916]::ty::context::TyCtxt, rustc_middle[64de8ef72bf26916]::ty::WithOptConstParam<rustc_span[ae34330c9831655e]::def_id::LocalDefId>, &rustc_data_structures[c1b71785fb17ed5f]::steal::Steal<rustc_middle[64de8ef72bf26916]::mir::Body>>
  76:     0x7fce726559fe - rustc_query_system[4a077ceeb940558c]::query::plumbing::try_execute_query::<rustc_query_impl[c278194eeae80ced]::plumbing::QueryCtxt, rustc_query_system[4a077ceeb940558c]::query::caches::DefaultCache<rustc_middle[64de8ef72bf26916]::ty::WithOptConstParam<rustc_span[ae34330c9831655e]::def_id::LocalDefId>, &rustc_data_structures[c1b71785fb17ed5f]::steal::Steal<rustc_middle[64de8ef72bf26916]::mir::Body>>>
  77:     0x7fce72647b0c - <rustc_query_impl[c278194eeae80ced]::Queries as rustc_middle[64de8ef72bf26916]::ty::query::QueryEngine>::mir_const
  78:     0x7fce71f4daf0 - rustc_mir_transform[b0e151ac016bcafc]::mir_promoted
  79:     0x7fce7311085c - <rustc_query_system[4a077ceeb940558c]::dep_graph::graph::DepGraph<rustc_middle[64de8ef72bf26916]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[64de8ef72bf26916]::ty::context::TyCtxt, rustc_middle[64de8ef72bf26916]::ty::WithOptConstParam<rustc_span[ae34330c9831655e]::def_id::LocalDefId>, (&rustc_data_structures[c1b71785fb17ed5f]::steal::Steal<rustc_middle[64de8ef72bf26916]::mir::Body>, &rustc_data_structures[c1b71785fb17ed5f]::steal::Steal<rustc_index[5483e2b3bbd55b6a]::vec::IndexVec<rustc_middle[64de8ef72bf26916]::mir::Promoted, rustc_middle[64de8ef72bf26916]::mir::Body>>)>
  80:     0x7fce72686c4d - rustc_query_system[4a077ceeb940558c]::query::plumbing::get_query::<rustc_query_impl[c278194eeae80ced]::queries::mir_promoted, rustc_query_impl[c278194eeae80ced]::plumbing::QueryCtxt>
  81:     0x7fce72648017 - <rustc_query_impl[c278194eeae80ced]::Queries as rustc_middle[64de8ef72bf26916]::ty::query::QueryEngine>::mir_promoted
  82:     0x7fce723e115f - rustc_borrowck[6560f87b90777ea6]::mir_borrowck
  83:     0x7fce723e0c4e - <rustc_borrowck[6560f87b90777ea6]::provide::{closure#0} as core[807c2f8b07026ed8]::ops::function::FnOnce<(rustc_middle[64de8ef72bf26916]::ty::context::TyCtxt, rustc_span[ae34330c9831655e]::def_id::LocalDefId)>>::call_once
  84:     0x7fce7311cacf - <rustc_query_system[4a077ceeb940558c]::dep_graph::graph::DepGraph<rustc_middle[64de8ef72bf26916]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[64de8ef72bf26916]::ty::context::TyCtxt, rustc_span[ae34330c9831655e]::def_id::LocalDefId, &rustc_middle[64de8ef72bf26916]::mir::query::BorrowCheckResult>
  85:     0x7fce72661b94 - rustc_query_system[4a077ceeb940558c]::query::plumbing::try_execute_query::<rustc_query_impl[c278194eeae80ced]::plumbing::QueryCtxt, rustc_query_system[4a077ceeb940558c]::query::caches::DefaultCache<rustc_span[ae34330c9831655e]::def_id::LocalDefId, &rustc_middle[64de8ef72bf26916]::mir::query::BorrowCheckResult>>
  86:     0x7fce726494aa - <rustc_query_impl[c278194eeae80ced]::Queries as rustc_middle[64de8ef72bf26916]::ty::query::QueryEngine>::mir_borrowck
  87:     0x7fce72093a00 - rustc_typeck[9089f5c415cb9e35]::collect::type_of::type_of
  88:     0x7fce725e73a3 - <rustc_query_system[4a077ceeb940558c]::dep_graph::graph::DepGraph<rustc_middle[64de8ef72bf26916]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[64de8ef72bf26916]::ty::context::TyCtxt, rustc_span[ae34330c9831655e]::def_id::DefId, rustc_middle[64de8ef72bf26916]::ty::Ty>
  89:     0x7fce726c2fbb - rustc_query_system[4a077ceeb940558c]::query::plumbing::get_query::<rustc_query_impl[c278194eeae80ced]::queries::type_of, rustc_query_impl[c278194eeae80ced]::plumbing::QueryCtxt>
  90:     0x7fce72080767 - rustc_typeck[9089f5c415cb9e35]::check::check::check_item_type
  91:     0x7fce72086b5e - rustc_typeck[9089f5c415cb9e35]::check::check::check_mod_item_types
  92:     0x7fce725e26f2 - <rustc_query_system[4a077ceeb940558c]::dep_graph::graph::DepGraph<rustc_middle[64de8ef72bf26916]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[64de8ef72bf26916]::ty::context::TyCtxt, rustc_span[ae34330c9831655e]::def_id::LocalDefId, ()>
  93:     0x7fce72664cbe - rustc_query_system[4a077ceeb940558c]::query::plumbing::try_execute_query::<rustc_query_impl[c278194eeae80ced]::plumbing::QueryCtxt, rustc_query_system[4a077ceeb940558c]::query::caches::DefaultCache<rustc_span[ae34330c9831655e]::def_id::LocalDefId, ()>>
  94:     0x7fce73218bf3 - rustc_query_system[4a077ceeb940558c]::query::plumbing::get_query::<rustc_query_impl[c278194eeae80ced]::queries::check_mod_item_types, rustc_query_impl[c278194eeae80ced]::plumbing::QueryCtxt>
  95:     0x7fce72f2329f - <rustc_middle[64de8ef72bf26916]::hir::map::Map>::for_each_module::<rustc_typeck[9089f5c415cb9e35]::check_crate::{closure#6}::{closure#0}>
  96:     0x7fce72f51be5 - rustc_typeck[9089f5c415cb9e35]::check_crate
  97:     0x7fce72c81a07 - rustc_interface[21c2a0e765b4b7f4]::passes::analysis
  98:     0x7fce7313a8ec - <rustc_query_system[4a077ceeb940558c]::dep_graph::graph::DepGraph<rustc_middle[64de8ef72bf26916]::dep_graph::dep_node::DepKind>>::with_task::<rustc_middle[64de8ef72bf26916]::ty::context::TyCtxt, (), core[807c2f8b07026ed8]::result::Result<(), rustc_errors[65e8676031e39f89]::ErrorGuaranteed>>
  99:     0x7fce731d870b - rustc_query_system[4a077ceeb940558c]::query::plumbing::try_execute_query::<rustc_query_impl[c278194eeae80ced]::plumbing::QueryCtxt, rustc_query_system[4a077ceeb940558c]::query::caches::DefaultCache<(), core[807c2f8b07026ed8]::result::Result<(), rustc_errors[65e8676031e39f89]::ErrorGuaranteed>>>
 100:     0x7fce732375ce - rustc_query_system[4a077ceeb940558c]::query::plumbing::get_query::<rustc_query_impl[c278194eeae80ced]::queries::analysis, rustc_query_impl[c278194eeae80ced]::plumbing::QueryCtxt>
 101:     0x7fce72c5a48e - <rustc_interface[21c2a0e765b4b7f4]::passes::QueryContext>::enter::<rustc_driver[45495022f976cfcf]::run_compiler::{closure#1}::{closure#2}::{closure#3}, core[807c2f8b07026ed8]::result::Result<(), rustc_errors[65e8676031e39f89]::ErrorGuaranteed>>
 102:     0x7fce72c3e1fe - <rustc_interface[21c2a0e765b4b7f4]::interface::Compiler>::enter::<rustc_driver[45495022f976cfcf]::run_compiler::{closure#1}::{closure#2}, core[807c2f8b07026ed8]::result::Result<core[807c2f8b07026ed8]::option::Option<rustc_interface[21c2a0e765b4b7f4]::queries::Linker>, rustc_errors[65e8676031e39f89]::ErrorGuaranteed>>
 103:     0x7fce72c3a4ea - rustc_span[ae34330c9831655e]::with_source_map::<core[807c2f8b07026ed8]::result::Result<(), rustc_errors[65e8676031e39f89]::ErrorGuaranteed>, rustc_interface[21c2a0e765b4b7f4]::interface::create_compiler_and_run<core[807c2f8b07026ed8]::result::Result<(), rustc_errors[65e8676031e39f89]::ErrorGuaranteed>, rustc_driver[45495022f976cfcf]::run_compiler::{closure#1}>::{closure#1}>
 104:     0x7fce72c3ecb0 - rustc_interface[21c2a0e765b4b7f4]::interface::create_compiler_and_run::<core[807c2f8b07026ed8]::result::Result<(), rustc_errors[65e8676031e39f89]::ErrorGuaranteed>, rustc_driver[45495022f976cfcf]::run_compiler::{closure#1}>
 105:     0x7fce72c6a542 - <scoped_tls[315da7db807c9876]::ScopedKey<rustc_span[ae34330c9831655e]::SessionGlobals>>::set::<rustc_interface[21c2a0e765b4b7f4]::interface::run_compiler<core[807c2f8b07026ed8]::result::Result<(), rustc_errors[65e8676031e39f89]::ErrorGuaranteed>, rustc_driver[45495022f976cfcf]::run_compiler::{closure#1}>::{closure#0}, core[807c2f8b07026ed8]::result::Result<(), rustc_errors[65e8676031e39f89]::ErrorGuaranteed>>
 106:     0x7fce72c3c8bf - std[9611d438937dc38]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[21c2a0e765b4b7f4]::util::run_in_thread_pool_with_globals<rustc_interface[21c2a0e765b4b7f4]::interface::run_compiler<core[807c2f8b07026ed8]::result::Result<(), rustc_errors[65e8676031e39f89]::ErrorGuaranteed>, rustc_driver[45495022f976cfcf]::run_compiler::{closure#1}>::{closure#0}, core[807c2f8b07026ed8]::result::Result<(), rustc_errors[65e8676031e39f89]::ErrorGuaranteed>>::{closure#0}, core[807c2f8b07026ed8]::result::Result<(), rustc_errors[65e8676031e39f89]::ErrorGuaranteed>>
 107:     0x7fce72c569e9 - <<std[9611d438937dc38]::thread::Builder>::spawn_unchecked_<rustc_interface[21c2a0e765b4b7f4]::util::run_in_thread_pool_with_globals<rustc_interface[21c2a0e765b4b7f4]::interface::run_compiler<core[807c2f8b07026ed8]::result::Result<(), rustc_errors[65e8676031e39f89]::ErrorGuaranteed>, rustc_driver[45495022f976cfcf]::run_compiler::{closure#1}>::{closure#0}, core[807c2f8b07026ed8]::result::Result<(), rustc_errors[65e8676031e39f89]::ErrorGuaranteed>>::{closure#0}, core[807c2f8b07026ed8]::result::Result<(), rustc_errors[65e8676031e39f89]::ErrorGuaranteed>>::{closure#1} as core[807c2f8b07026ed8]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
 108:     0x7fce702926c3 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h269b15882926c1ec
                               at /rustc/6dbae3ad19309bb541d9e76638e6aa4b5449f29a/library/alloc/src/boxed.rs:1935:9
 109:     0x7fce702926c3 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h58323387bc1c2642
                               at /rustc/6dbae3ad19309bb541d9e76638e6aa4b5449f29a/library/alloc/src/boxed.rs:1935:9
 110:     0x7fce702926c3 - std::sys::unix::thread::Thread::new::thread_start::h916fd4de50349453
                               at /rustc/6dbae3ad19309bb541d9e76638e6aa4b5449f29a/library/std/src/sys/unix/thread.rs:108:17
 111:     0x7fce70190ea7 - start_thread
 112:     0x7fce700bedef - clone
 113:                0x0 - <unknown>
```

</p>
</details>

**Comment by compiler-errors** [comment]
I think this is fixed by #99928

**Comment by compiler-errors** [comment]
I'm pretty sure this is fixed, please re-open an issue if you see this again, since I am unable to test the incomplete code you provided.
